distribution/lib/Standard/Test/0.0.0-dev/src/Suite.enso

```diff
@@ -61,46 +61,55 @@ type Suite
     run_with_filter self (filter : (Text | Nothing) = Nothing) (should_exit : Boolean = True) -> (Boolean | Nothing) =
         config = Suite_Config.from_environment
 
-        # Map of groups to vector of specs that match the filter
-        matching_specs = self.groups.fold Map.empty map-> group->
+        # List of pairs of groups and their specs that match the filter
+        matching_specs = self.groups.flat_map group->
             group_matches = name_matches group.name filter
             case group_matches of
                 True ->
                     # Include all the specs from the group
-                    map.insert group group.specs
+                    [[group, group.specs]]
                 False ->
                     # Try to include only some specs from the group
                     matched_specs = group.specs.filter spec->
                         name_matches spec.name filter
                     case matched_specs.is_empty of
-                        True -> map
+                        True -> []
                         False ->
-                            assert (map.contains_key group . not)
-                            map.insert group matched_specs
+                            [[group, matched_specs]]
 
+        # If ANSI colors are enabled, we assume an interactive terminal and will print progress
+        progress_reporter = case config.use_ansi_colors of
```

Comment:
This is a bit smelly: I'm overriding the meaning of `use_ansi_colors` to generally mean that the application is run in an 'interactive' terminal session.

Shall we rename this setting to `interactive_terminal`? It also does not have the same meaning.

We could introduce a _separate_ setting for this but I think that is adding unnecessary complication - in general both colors and progress bars are going to be used in the same setting - on a dev machine, but not on CI.

Reply:
`interactive_terminal` definitely sounds more appropriate.

> in general both colors and progress bars are going to be used in the same setting - on a dev machine, but not on CI.

Colors **are used on the CI**. And I think that in helps with readability. At least it helps to me. I would like to retain the colors on the CI, and not print the progress bar on the CI.

Reply:
Oh, I was somehow sure that colors are _not_ used on CI. My mistake. Of course the progress bar should not run on CI. I will try a different approach then.